theano/tensor/subtensor.py

```diff
@@ -1408,6 +1408,10 @@ def c_code(self, node, name, inputs, outputs, sub):
         {
             Py_XDECREF(%(z)s);
             %(z)s = %(copy_of_x)s;
+            if (!%(z)s) {
+                PyErr_SetString(PyExc_MemoryError, "Could not allocate copy of input.");
```

Comment:
I am reasonably certain that the CPU code already sets an exception and 100% certain that the GPU code does.